docker/containerbuild.sh

```diff
@@ -32,7 +32,17 @@ if [ -d "$BUILDROOT" ]; then
 fi
 
 # Fetch the boot.iso for the build.
-curl http://mirror.centos.org/centos/"${KSNAME##*-}"/os/x86_64/images/boot.iso -o /tmp/boot-"${KSNAME##*-}".iso
+if [[ ${CONT_ARCH} -eq "ppc64le" ]]; then
```

Comment:
What if a new architecture needs to be handled?

I propose we use an associative array, like a dictionary, where the index is formed by the version plus the architecture.  So we can have a dictionary of boot.iso URLs for each pair of version+architecture.

Something like the following:

```
$ git diff
diff --git a/docker/containerbuild.sh b/docker/containerbuild.sh
index 3744320..d7d993c 100755
--- a/docker/containerbuild.sh
+++ b/docker/containerbuild.sh
@@ -10,10 +10,23 @@
 USAGE="USAGE: $(basename "$0") kickstart"
 KICKSTART="$1"
 KSNAME=${KICKSTART%.*}
+KSVER=$(expr "${KSNAME}" : '.*\-\([0-9\.]*\).*')
 BUILDDATE=$(date +%Y%m%d)
 BUILDROOT=/var/tmp/containers/$BUILDDATE/$KSNAME
 CONT_ARCH=$(uname -m)
 
+declare -A BOOTISO_URLS
+BOOTISO_URLS=( \
+    ["5.11x86_64"]="http://mirror.centos.org/centos/5.11/os/x86_64/images/boot.iso" \
+    ["5x86_64"]="http://mirror.centos.org/centos/5/os/x86_64/images/boot.iso" \
+    ["6x86_64"]="http://mirror.centos.org/centos/6/os/x86_64/images/boot.iso" \
+    ["7arm64"]="TBD" \
+    ["7ppc64le"]="$(curl -L http://mirror.centos.org/altarch/7/isos/ppc64le 2>&1 | sed -n -r -s 's,.*(CentOS.*NetInstall.*\.iso).*,http://mirror.centos.org/altarch/7/isos/ppc64le/\1,gp')" \
+    ["7x86_64"]="http://mirror.centos.org/centos/7/os/x86_64/images/boot.iso" \
+)
+BOOTISO_URL=${BOOTISO_URLS[${KSVER}${CONT_ARCH}]}
+BOOTISO_PATH=/var/tmp/centos-boot-${KSVER}${CONT_ARCH}.iso
+
 #### Test for script requirements
 # Did we get passed a kickstart
 if [ "$#" -ne 1 ]; then
@@ -32,10 +45,10 @@ if [ -d "$BUILDROOT" ]; then
 fi
 
 # Fetch the boot.iso for the build.
-curl http://mirror.centos.org/centos/"${KSNAME##*-}"/os/x86_64/images/boot.iso -o /tmp/boot-"${KSNAME##*-}".iso
+curl ${BOOTISO_URL} -o ${BOOTISO_PATH}
 
 # Build the rootfs
-time livemedia-creator --logfile=/tmp/"$KSNAME"-"$BUILDDATE".log --make-tar --ks "$KICKSTART" --image-name="$KSNAME"-docker.tar.xz  --iso /tmp/boot-"${KSNAME##*-}".iso
+time livemedia-creator --logfile=/tmp/${KSNAME}-${BUILDDATE}.log --make-tar --ks "${KICKSTART}" --image-name=${KSNAME}-docker.tar.xz  --iso ${BOOTISO_PATH}
 
 # Put the rootfs someplace
 mkdir -p $BUILDROOT/docker
```

In this approach, the version of the netinstall image for ppc64le is not hardcoded.

To handle a new version and architecture, it may be as easy as updating indexes for the BOOTISO_URLS array.